### PR TITLE
Report data loading errors

### PR DESCRIPTION
**Pull request by lassoan**
Implement error reporting for loading data (similarly how it was recently implemented for *saving* data) and a two other minor fixes.

image

### CODE REVIEW

File: Libs/MRML/DisplayableManager/vtkMRMLViewInteractorStyle.cxx [hunk 1]
```diff
@@ -366,9 +366,13 @@ bool vtkMRMLViewInteractorStyle::DelegateInteractionEventDataToDisplayableManage
...
-    //this->SetMouseCursor(VTK_CURSOR_DEFAULT);
-    this->DisplayableManagers->GetNthDisplayableManager(0)->SetMouseCursor(VTK_CURSOR_DEFAULT);
+    // None of the displayable managers can process the event, just ignore it.
+    // If click events (non-keyboard events) cannot be processed here then
+    // indicate this by setting the mouse cursor to default.
+    if (eventData->GetType() != vtkCommand::KeyPressEvent && eventData->GetType() != vtkCommand::KeyReleaseEvent)
+      {
+      this->DisplayableManagers->GetNthDisplayableManager(0)->SetMouseCursor(VTK_CURSOR_DEFAULT);
+      }
     return false;
```
Comment (pieper) [inline Libs/MRML/DisplayableManager/vtkMRMLViewInteractorStyle.cxx:376]: Looks like a useful cleanup, but this is unrelated to the file io status.

File: Modules/Loadable/Markups/VTKWidgets/vtkSlicerCurveWidget.cxx [hunk 2]
```diff
@@ -38,6 +38,9 @@ vtkSlicerCurveWidget::vtkSlicerCurveWidget()
   this->SetEventTranslationClickAndDrag(WidgetStateOnWidget, vtkCommand::RightButtonPressEvent, vtkEvent::AltModifier,
     WidgetStateScale, WidgetEventScaleStart, WidgetEventScaleEnd);
 
+  // Accept Ctrl+MouseMove (and process as simple mouse move) so that this widget keeps the focus when the user moves
+  // the mouse while holding down Ctrl key for inserting a point.
+  this->SetEventTranslation(WidgetStateOnWidget, vtkCommand::MouseMoveEvent, vtkEvent::ControlModifier, WidgetEventMouseMove);
```
Comment (pieper) [inline Modules/Loadable/Markups/VTKWidgets/vtkSlicerCurveWidget.cxx:43]: Another change unrelated to file io

**Review by pieper — APPROVED** [review]
A couple unrelated changes - they look fine so it's okay with me to merge them with this PR or separate them out if you want.

**Comment by pieper** [comment]
This is a really nice addition - thanks!

**Comment by lassoan** [comment]
Thanks for the review, I'll separate those two unrelated changes into new commits.

**Comment by lassoan** [comment]
I've checked and the two unrelated changes are in separate commits already.

I've just started to not push changes directly to the Slicer repository anymore so that I work the same way as all other non-admin developers. I did not want to create a new pull request for the two unrelated small fixes because I did not want to add too much noise, but maybe in the future I'll create separate pull request for each commit, no matter how small they are.

File: Libs/MRML/Core/vtkMRMLMessageCollection.cxx [hunk 3]
```diff
@@ -23,6 +23,11 @@
 // MRML includes
 #include "vtkCommand.h"
 
+namespace
+{
+  const int SEPARATOR_MESSAGE_TYPE = vtkCommand::PropertyModifiedEvent;
```
Comment (jcfr) [inline Libs/MRML/Core/vtkMRMLMessageCollection.cxx:28]: Is the use of `vtkCommand::PropertyModifiedEvent` arbitrary ?

Comment (lassoan) [reply Libs/MRML/Core/vtkMRMLMessageCollection.cxx:28]: Yes, completely arbitrary, I just chose an event ID that has a name that makes some sense (separator is when we switch to another group, which is kind of moving to a different property) and not likely to be used otherwise.

Comment (lassoan) [reply Libs/MRML/Core/vtkMRMLMessageCollection.cxx:28]: I'll add a comment to explain this in a follow-up commit.
